spec/classes/init_spec.rb

```diff
@@ -263,6 +264,14 @@
           it { is_expected.not_to create_service('systemd-resolved').with_enable(true) }
           it { is_expected.not_to create_service('systemd-networkd').with_ensure('running') }
           it { is_expected.not_to create_service('systemd-networkd').with_enable(true) }
+
+          if facts[:os]['family'] == 'Debian'
+            if Puppet::Util::Package.versioncmp(facts[:os]['release']['full'], '20.04') >= 0 || Puppet::Util::Package.versioncmp(facts[:os]['release']['major'], '11') >= 0
```

Comment:
You could even make sure it's not set on non-Debian:
```suggestion
          if facts[:os]['family'] == 'Debian' && (Puppet::Util::Package.versioncmp(facts[:os]['release']['full'], '20.04') >= 0 || Puppet::Util::Package.versioncmp(facts[:os]['release']['major'], '11') >= 0)
```

Reply:
As my condition wasn't working, I've changed it.